File: api-xmlsignature/src/tr/gov/tubitak/uekae/esya/api/xmlsignature/XMLSignatureException.java
package tr.gov.tubitak.uekae.esya.api.xmlsignature;

import tr.gov.tubitak.uekae.esya.api.signature.SignatureException;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.util.I18n;

/**
 * @author ahmety
 * date: May 8, 2009
 */
public class XMLSignatureException extends SignatureException
{

    public XMLSignatureException(String aMessageId)
    {
        super(I18n.translate(aMessageId));
    }

    public XMLSignatureException(Throwable aCause, String aMessageId)
    {
        super(I18n.translate(aMessageId), aCause);
    }

    public XMLSignatureException(String aMessageId, Object... aMessageArgs)
    {
        super(I18n.translate(aMessageId, aMessageArgs));
    }

    public XMLSignatureException(Throwable aCause, String aMessageId, Object... aMessageArgs)
    {
        super(I18n.translate(aMessageId, aMessageArgs), aCause);
    }
}

File: api-xmlsignature/csharp/src/XMLSignatureException.cs
using System;
using tr.gov.tubitak.uekae.esya.api.signature;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature
{
	using I18n = tr.gov.tubitak.uekae.esya.api.xmlsignature.util.I18n;

	/// <summary>
	/// @author ahmety
	/// date: May 8, 2009
	/// </summary>
    public class XMLSignatureException : SignatureException
	{

		public XMLSignatureException(string aMessageId)
            : base(I18n.translate(aMessageId))
		{
		}

		public XMLSignatureException(Exception aCause, string aMessageId)
            : base(I18n.translate(aMessageId), aCause)
		{
		}

		public XMLSignatureException(string aMessageId, params object[] aMessageArgs)
            : base(I18n.translate(aMessageId, aMessageArgs))
		{
		}

        public XMLSignatureException(Exception aCause, string aMessageId, params object[] aMessageArgs)
            : base(I18n.translate(aMessageId, aMessageArgs), aCause)
		{
		}
	}

}